<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/tw017/Desktop/PipetteDetect/catkin_ws/src/core_moveit_config/urdf/dual_arm_robot.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="dual_arm_robot">
    <!-- <!-\-ros_control -->
    <!-- plugin-\-> -->
    <!-- <gazebo> -->
    <!--   <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so" /> -->
    <!-- </gazebo> -->
    <!-- <!-\-joint_state_publisher -->
    <!-- plugin-\-> -->
    <!-- <gazebo> -->
    <!--   <plugin name="joint_state_publisher" filename="libgazebo_ros_joint_state_publisher.so"> -->
    <!--     <jointName>arm_joint_0, arm_joint_1, arm_joint_2, arm_joint_3, arm_joint_4, arm_joint_5</jointName> -->
    <!--   </plugin> -->
    <!-- </gazebo> -->
    <link name="world" />
    <joint name="mycobot_world2base" type="fixed">
        <parent link="world" />
        <child link="mycobot_base_link" />
        <origin rpy="0 0 0" xyz="0 0 0" />
    </joint>
    <link name="mycobot_base_link">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/base.dae" />
            </geometry>
            <origin rpy=" 0 0 3.1415926" xyz="0 0 0 " />
        </visual>
        <collision>
            <origin rpy=" 0 0 3.1415926" xyz="0 0 0.05 " />
            <geometry>
                <cylinder length="0.100" radius="0.0725" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1.0" />
            <inertia ixx="0.00218" ixy="0.0" ixz="0.0" iyy="0.00218" iyz="0.0" izz="0.00263" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link1">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link1_1.dae" />
            </geometry>
            <origin rpy=" 0 0 0" xyz="0 0 0 " />
        </visual>
        <collision>
            <origin rpy=" 0 0 0" xyz="0 0 -0.02 " />
            <geometry>
                <cylinder length="0.106" radius="0.043" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.5" />
            <inertia ixx="0.00070" ixy="0.0" ixz="0.0" iyy="0.00070" iyz="0.0" izz="0.00046" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link2">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link2.dae" />
            </geometry>
            <origin rpy=" 0 0 3.1415926" xyz="0 0 0.0587 " />
        </visual>
        <collision>
            <origin rpy="0 -1.5708 0" xyz="0.07 0 0.095 " />
            <geometry>
                <cylinder length="0.18" radius="0.036" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.7" />
            <inertia ixx="0.00212" ixy="0.0" ixz="0.0" iyy="0.00212" iyz="0.0" izz="0.00045" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link3">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link3.dae" />
            </geometry>
            <origin rpy=" 0 0 3.1415926" xyz="0 0 0.038 " />
        </visual>
        <collision>
            <origin rpy=" 0 -1.5708 0" xyz="0.044 0 0 " />
            <geometry>
                <cylinder length="0.14" radius="0.0305" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.5" />
            <inertia ixx="0.00093" ixy="0.0" ixz="0.0" iyy="0.00093" iyz="0.0" izz="0.00023" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link4">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link4.dae" />
            </geometry>
            <origin rpy=" 0 0 0" xyz="0 -0.001 0 " />
        </visual>
        <collision>
            <origin rpy=" 0 -1.5708 1.5708" xyz="0.0 -0.014 0 " />
            <geometry>
                <cylinder length="0.083" radius="0.029" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.2" />
            <inertia ixx="0.00016" ixy="0.0" ixz="0.0" iyy="0.00016" iyz="0.0" izz="0.00008" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link5">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link5.dae" />
            </geometry>
            <origin rpy=" 0  0 3.1415926" xyz="0 0.002  0 " />
        </visual>
        <collision>
            <origin rpy=" 1.5708 1.5708 0" xyz="0 -0.01 0 " />
            <geometry>
                <cylinder length="0.0849" radius="0.029" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.2" />
            <inertia ixx="0.00016" ixy="0.0" ixz="0.0" iyy="0.00016" iyz="0.0" izz="0.00008" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <link name="mycobot_link6">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_320_moveit/urdf/320_urdf/link6.dae" />
            </geometry>
            <material name="grey">
                <color rgba="0.5 0.5 0.5 1" />
            </material>
            <origin rpy=" 0 1.5708 0" xyz="-0.502 -0.090 -0.096 " />
        </visual>
        <collision>
            <origin rpy=" 0 0 0" xyz="0.0 0 -0.01 " />
            <geometry>
                <cylinder length="0.0225" radius="0.026" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.2" />
            <inertia ixx="0.00002" ixy="0.0" ixz="0.0" iyy="0.00002" iyz="0.0" izz="0.00003" />
            <origin xyz="0 0 0" rpy="0 0 0" />
        </inertial>
    </link>
    <joint name="mycobot_arm_joint_0" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_base_link" />
        <child link="mycobot_link1" />
        <origin rpy="0 0 0" xyz="0 0 0.17846" />
    </joint>
    <joint name="mycobot_arm_joint_1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_link1" />
        <child link="mycobot_link2" />
        <origin rpy="0 -1.5708 1.5708" xyz="0 0 0" />
    </joint>
    <joint name="mycobot_arm_joint_2" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_link2" />
        <child link="mycobot_link3" />
        <origin rpy="0 0 0" xyz="0.13635 0 0   " />
    </joint>
    <joint name="mycobot_arm_joint_3" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_link3" />
        <child link="mycobot_link4" />
        <origin rpy=" 0 0 1.57080" xyz="0.1 0 0.09 " />
    </joint>
    <joint name="mycobot_arm_joint_4" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_link4" />
        <child link="mycobot_link5" />
        <origin rpy="1.5708 -1.5708 0" xyz="0 -0.07318 0" />
    </joint>
    <joint name="mycobot_arm_joint_5" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="100.0" lower="-3.14" upper="3.14159" velocity="5" />
        <parent link="mycobot_link5" />
        <child link="mycobot_link6" />
        <origin rpy="-1.5708 0 0" xyz="0.00 0.06635 0" />
    </joint>
    <transmission name="mycobot_arm_tran_0">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_0">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_0">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="mycobot_arm_tran_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_1">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="mycobot_arm_tran_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_2">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="mycobot_arm_tran_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_3">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="mycobot_arm_tran_4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_4">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="mycobot_arm_tran_5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="mycobot_arm_joint_5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="mycobot_arm_motor_5">
            <hardwareInterface>PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
            <robotNamespace>/cobotta</robotNamespace>
        </plugin>
    </gazebo>
    <!-- <link name="world" /> -->
    <joint name="joint_w" type="fixed">
        <parent link="world" />
        <child link="base_link" />
        <origin rpy="0 0 0" xyz="0 0.3 0" />
    </joint>
    <link name="base_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/base_link.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/base_link.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <link name="J1">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J1.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J1.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_1" type="revolute">
        <parent link="base_link" />
        <child link="J1" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.108000" />
        <axis xyz="-0.000000 -0.000000 1.000000" />
        <limit effort="1" lower="-2.61799387799149" upper="2.61799387799149" velocity="0.383012504350156" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <link name="J2">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J2.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J2.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_2" type="revolute">
        <parent link="J1" />
        <child link="J2" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.072000" />
        <axis xyz="-0.000000 1.000000 -0.000000" />
        <limit effort="1" lower="-1.0471975511966" upper="1.74532925199433" velocity="0.37256670877697" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <link name="J3">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J3.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J3.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_3" type="revolute">
        <parent link="J2" />
        <child link="J3" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.165000" />
        <axis xyz="-0.000000 1.000000 -0.000000" />
        <limit effort="1" lower="0.314159265358979" upper="2.44346095279206" velocity="0.68942250783028" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <link name="J4">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J4.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J4.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_4" type="revolute">
        <parent link="J3" />
        <child link="J4" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="-0.012000 0.020000 0.088000" />
        <axis xyz="-0.000000 -0.000000 1.000000" />
        <limit effort="1" lower="-2.96705972839036" upper="2.96705972839036" velocity="0.702189591308619" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <link name="J5">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J5.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J5.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_5" type="revolute">
        <parent link="J4" />
        <child link="J5" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 -0.020000 0.089500" />
        <axis xyz="-0.000000 1.000000 -0.000000" />
        <limit effort="1" lower="-1.65806278939461" upper="2.35619449019234" velocity="0.702189591308619" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <link name="J6">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J6.dae" scale="1 1 1" />
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://denso_robot_descriptions/cobotta_description/J6.dae" scale="1 1 1" />
            </geometry>
        </collision>
        <inertial>
            <mass value="1" />
            <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
    </link>
    <joint name="joint_6" type="revolute">
        <parent link="J5" />
        <child link="J6" />
        <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 -0.044500 0.042000" />
        <axis xyz="-0.000000 -0.000000 1.000000" />
        <limit effort="1" lower="-2.96705972839036" upper="2.96705972839036" velocity="1.05386470893922" />
        <dynamics damping="0" friction="0" />
    </joint>
    <transmission name="trans_6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_6">
            H
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor_6">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control" />
        <robotNamespace>/cobotta</robotNamespace>
    </gazebo>
    <plugin filename="libgazebo_ros_joint_state_publisher.so" name="mycobot_joint_state_publisher">
        <robotNamespace>/mycobot</robotNamespace>
        <jointName>mycobot_arm_joint_0, mycobot_arm_joint_1, mycobot_arm_joint_2, mycobot_arm_joint_3, mycobot_arm_joint_4, mycobot_arm_joint_5</jointName>
    </plugin>
</robot>

